subt/tools/analyze_detectors.py

```diff
@@ -0,0 +1,243 @@
+                    'robot': 1.0,
+                    'breadcrumb': 1.0,
+                    'nothing': 1.0
+                    }
+MIN_SCORES = np.array(list(NAMES_AND_SCORES.values()))
+
+
+def take_third(item):
+    return item[2]
+
+
+def proces_torch_result(result):
+    result.sort(key=take_third, reverse=True)
+    score_torch = result[1][2]
+    points = [[x, y] for x, y, c in result][:2]
+
+    return score_torch, points
+
+
+def manual_sorting(data):
+    ii = 0
+    while True:
+        if ii < 0:
+            ii = 0
+        if ii >= len(data):
+            ii = len(data) - 1
+        artf_name, score_t, score_cv, points, bbox, im_path, detection_type = data[ii]
+        print(bbox)
+        x, y, xw, yh = bbox
+        im_path = im_path[:-1]  # remove line end
+        img = cv2.imread(str(im_path), 1)
+        assert img is not None
+        cv2.rectangle(img, (x, y), (xw, yh), (0, 0, 255))
+        cv2.putText(img, artf_name, (x, y), fontFace=cv2.FONT_HERSHEY_SIMPLEX, fontScale=1, color=(0, 0, 255))
+        for xc, yc in points:
+            cv2.circle(img, (xc, yc), radius=4, color=(255, 255, 0), thickness=1)
+        cv2.putText(img, str(detection_type), (10, 50), fontFace=cv2.FONT_HERSHEY_SIMPLEX, fontScale=1, color=(255, 0, 0))
+        im_to_show = cv2.resize(img, (1280, 720))
+        cv2.imshow("win", im_to_show)
+
+        k = cv2.waitKey(0) & 0xFF
+        if k == ord("n"):  # next img
+            ii += 1
+        elif k == ord("b"):  # back one img
+            ii -= 1
+        elif k == ord("t"):
+            data[ii][6] = "true"  # true detection
+        elif k == ord("f"):
+            data[ii][6] = "false"  # false detection
+        elif k == ord("d"):
+            data[ii][3] = "None"  # do not use
+        elif k == ord("q"):  # close and save
+            break
+
+    cv2.destroyAllWindows()
+
+    return [[artf_name, score_t, score_cv, detection_type] for artf_name, score_t, score_cv, __, __, __, detection_type in data]
+
+
+def create_detector():
+    model = os.path.join(os.path.dirname(__file__), '../../../mdnet5.128.128.13.4.elu.pth')
+    confidence_thresholds = {
+        'survivor': 0.5,
+        'backpack': 0.5,
+        'phone': 0.5,
+        'helmet': 0.5,
+        'rope': 0.5,
+        'fire_extinguisher': 0.5,
+        'drill': 0.5,
+        'vent': 0.5,
+        'cube' : 0.5
+    }
+    max_gap = 16
+    min_group_size = 2
+
+    use_cuda = torch.cuda.is_available()
+    device = torch.device("cuda" if use_cuda else "cpu")
+    print('Using:', device)
+    model, categories = subt.artf_model.load_model(model, device)
+    return Detector(model, confidence_thresholds, categories, device,
+                    max_gap, min_group_size)
+
+
+def log_eval(log_file):
+    data_dir = log_file + ".d"
+    os.makedirs(data_dir,exist_ok=True)
+    detection_log = open(os.path.join(data_dir, "detection.log"), "w")
+
+    detector = create_detector()
+    CvDetector().min_score = MIN_SCORES
```

Comment:
What does CvDetector do? It returns a global singleton?
(I am trying to understand the two lines here. They look unusual to me.)

Reply:
There is default variable `min_score` (like `confidence_thresholds`) and this variable have to be changed because we loking for false detections. Should i change my way?